monitor/xfeemngr/gasprice/gasprice_test.go

```diff
@@ -0,0 +1,107 @@
+package gasprice_test
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/omni-network/omni/monitor/xfeemngr/gasprice"
+
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/stretchr/testify/require"
+)
+
+type mockPricer struct {
+	mu      sync.RWMutex
+	price   uint64
+	initial uint64
+}
+
+func newMock(price uint64) *mockPricer {
+	return &mockPricer{
+		price:   price,
+		initial: price,
+	}
+}
+
+func (m *mockPricer) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	return big.NewInt(int64(m.price)), nil
+}
+
+func (m *mockPricer) setPrice(price uint64) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.price = price
+}
+
+func TestBufferStream(t *testing.T) {
+	t.Parallel()
+
+	mocks := map[uint64]*mockPricer{
+		1: newMock(100),
+		2: newMock(200),
+		3: newMock(300),
+	}
+
+	pricers := map[uint64]ethereum.GasPricer{
+		1: mocks[1],
+		2: mocks[2],
+		3: mocks[3],
+	}
+
+	thresh := 0.1
+	interval := 100 * time.Millisecond
+
+	b := gasprice.NewBuffer(pricers, gasprice.WithThresholdPct(thresh), gasprice.WithStreamInterval(interval))
+	b.Stream(context.Background())
+
+	time.Sleep(5 * interval)
```

Comment:
I don't like time sleeps in tests, TBH. Ideally, we need some Clock interface, which implements the `Now()` method. Then we can adjust the clock back and forth instead of having to sleep and wait for something to happen.

Reply:
Yeah I also prefer not to time.Sleep in tests but did not think there was a way around int

Reply:
I can refactor out the ticker

Reply:
yeah that's the plan - but refactoring it out so that I can use a mock ticker

Reply:
done